internal/models/sessions.go

```diff
@@ -94,10 +99,13 @@ func FindSessionByID(tx *storage.Connection, id uuid.UUID, forUpdate bool) (*Ses
 		// queries which lock the rows affected by the query from
 		// being accessed by any other transaction that also uses FOR
 		// UPDATE
-		if err := tx.RawQuery(fmt.Sprintf("SELECT id FROM %q WHERE id = ? LIMIT 1 FOR UPDATE;", session.TableName()), id).Exec(); err != nil {
+		if err := tx.RawQuery(fmt.Sprintf("SELECT * FROM %q WHERE id = ? LIMIT 1 FOR UPDATE SKIP LOCKED;", session.TableName()), id).First(session); err != nil {
```

Comment:
Do you mind expanding more about this change?  Wondering why we need to move from selecting the id to selecting the entire row. Does it activate a tuple/row lock?

Reply:
So previously we were just interested in locking the row and didn't care about what was selected. But now, we actually do care about _what_ was selected. Because we use `SKIP LOCKED`, if the row is locked, the query will return "no rows." So we have to pretend we're pop and use `.First()` to try and get the row that was selected.